packages/common/core-utils/src/lazy.ts

```diff
@@ -45,7 +45,9 @@ export class Lazy<T> {
  */
 export class LazyPromise<T> implements Promise<T> {
 	public get [Symbol.toStringTag](): string {
-		return this.getPromise()[Symbol.toStringTag];
+		return this.result === undefined
```

Comment:
I know I suggested this change to begin with, but to keep it simpler I'm leaning towards just always returning `"[object LazyPromise]"`